src/EnergyPlus/api/datatransfer.py

```diff
@@ -608,8 +607,6 @@ def get_construction_handle(self, state: c_void_p, var_name: Union[str, bytes])
         :param var_name: Name of the construction to look up
         :return: An integer ID for this construction, or -1 if one could not be found.
         """
-        if not self.running_as_python_plugin:
-            raise EnergyPlusException("get_construction_handle is only available as part of a Python Plugin workflow")
```

Comment:
Actual fix for #9957

Reply:
OK.  I don't recall exactly why I only had this in place for plugin applications.  But if it's working in your tests, I'm happy with this.